packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -156,8 +171,49 @@ const clampTimeRange = (timeRange: ITimeRange): ITimeRange => {
   };
 };
 
-const generateDefaultTime = (increments: number, timeRange: ITimeRange | undefined) => {
-  const newDefaultTime = new Date();
+const getDateFromSelection = (showSeconds: boolean, useHour12: boolean, defaultTime: Date, selectedTime: string) => {
+  const splitValue = selectedTime.split(':');
+  let hours = +splitValue[0];
+  let minutes;
+  let seconds = 0;
+  let ap;
+  if (showSeconds) {
+    minutes = +splitValue[1];
+    seconds = +splitValue[2].split(' ')[0];
+    if (useHour12) {
+      ap = splitValue[2].split(' ')[1];
+    }
+  } else {
+    minutes = +splitValue[1].split(' ')[0];
+    if (useHour12) {
+      ap = splitValue[1].split(' ')[1];
+    }
+  }
+
+  if (useHour12 && ap && ap.toLowerCase() === 'pm' && hours !== 12) {
+    hours += TimeConstants.OffsetTo24HourFormat;
+  }
+
+  let hoursOffset;
+  if (useHour12 && defaultTime.getHours() >= hours) {
+    hoursOffset = TimeConstants.HoursInOneDay - defaultTime.getHours() + hours;
+  } else {
+    hoursOffset = Math.abs(defaultTime.getHours() - hours);
+  }
+
+  const offset =
+    TimeConstants.MillisecondsIn1Sec * TimeConstants.MinutesInOneHour * hoursOffset * TimeConstants.SecondsInOneHour +
+    seconds;
```

Comment:
`seconds` is added to some milliseconds computed in line 205, I think this should be
```suggestion
    seconds * TimeConstants.MillisecondsIn1Sec;
```

Reply:
Resolved by adding `* TimeConstants.MillisecondsIn1Sec` to calculation.